datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -657,17 +657,15 @@ macro_rules! compute_utf8_op_scalar {
 
 /// Invoke a compute kernel on a data array and a scalar value
 macro_rules! compute_utf8_op_dyn_scalar {
-    ($LEFT:expr, $RIGHT:expr, $OP:ident) => {{
+    ($LEFT:expr, $RIGHT:expr, $OP:ident, $OP_TYPE:expr) => {{
         if let Some(string_value) = $RIGHT {
             Ok(Arc::new(paste::expr! {[<$OP _dyn_utf8_scalar>]}(
                 $LEFT,
                 &string_value,
             )?))
         } else {
-            Err(DataFusionError::Internal(format!(
-                "compute_utf8_op_scalar for '{}' failed with literal 'none' value",
-                stringify!($OP),
-            )))
+            // when the $RIGHT is a NULL, generate a NULL array of $OP_TYPE
+            Ok(Arc::new(new_null_array($OP_TYPE, $LEFT.len())))
```

Comment:
1. I think we should respect `SortOptions` here as well for ScalaValue.
2. Question: how about the `NULL != column` case, do we have an expression normalizer that asserts `Null's often coming as the right-hand op for a binary comparison?

Reply:
@yjshen  what do you mean `SortOptions`?  I don't think these expressions are used directly in ORDER BY clauses -- they are used instead for expressions like `a < b` 

Adding some more tests in `sql/expr.rs`  for `NULL < column` and `NULL > column` sounds like a good idea.

Reply:
After checking Postgres and MySQL, I found I'm wrong on the expected behavior while comparing columns with NULL using `<` `>` ..., they will return NULL in all cases. I was expecting to apply null first or null last while comparing column with NULL, so always true or always false at the first time.

Reply:
Thanks @alamb @yjshen 
I agree this macro is just used in expression `evaluate`, maybe we can ignore it in this pr.
I'll look into this case `NULL != column` carefully and add more cases in ut.
Besides, I am super busy these days, I'll do it ASAP 🤟

Reply:
I will spend some time writing the `NULL != column`

Reply:
Follow on PR: https://github.com/apache/arrow-datafusion/pull/2510# turns out there was a bug!